format.go

```diff
@@ -55,9 +60,27 @@ func (f *Formatter) Format(o interface{}, fields []string) (interface{}, error)
 		}
 		f.formatters[t][key] = ff
 	}
-	v, err := ff.format(v)
+	v, err = ff.format(v)
 	if err != nil {
 		return nil, err
 	}
 	return v.Interface(), nil
 }
+
+// detectDuplicateFields returns an error if passed the same field more than once.
+func detectDuplicateFields(fields []string) error {
+	h := make(map[string]int)
+	for _, f := range fields {
+		h[f]++
+	}
+	e := []string{}
```

Comment:
`var e []string` is more efficient (the slice is only allocated by the first call to append), and `len([]string(nil)) == 0`